File: 2025-04-15/Invert_Binary_Tree.java
public class Invert_Binary_Tree {
    public static TreeNode invertTree(TreeNode root) {
        if(root == null) return null;

        TreeNode node = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(node);

        return root;
    }
    public static void main(String[] args) {
        Integer[] arr = new Integer[]{4,2,7,1,3,6,9};
        Tree tree = new Tree();
        TreeNode root = tree.arrayToTree(arr);
        TreeNode ans = invertTree(root);
        tree.printTree(ans);
    }
}

File: 2026-02-12/Invert_Binary_Tree.py
from typing import Optional, List
from Tree import TreeNode, Tree


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None

        root.left, root.right = root.right, root.left

        self.invertTree(root.left)
        self.invertTree(root.right)

        return root


sol = Solution()
tree = Tree()
node = tree.array_to_tree([4,2,7,1,3,6,9])
print(sol.invertTree(node))
